commodore/catalog.py

```diff
@@ -69,6 +69,49 @@ def clean_catalog(repo):
         rm_tree_contents(repo.working_tree_dir)
 
 
+def _push_catalog(cfg: Config, repo, commit_message):
+    """Push catalog to catalog repo if conditions to allow push are met.
+
+    Conditions to allow pushing are:
+    * Commodore doesn't run in local mode
+    * User has requested pushing with `--push`
+
+    Ask user to confirm push if `--interactive` is specified
+    """
+    if not cfg.local:
+        if cfg.interactive and cfg.push:
+            cfg.push = click.confirm(" > Should the push be done?")
+
+        if cfg.push:
+            click.echo(" > Commiting changes...")
+            git.commit(repo, commit_message, cfg)
```

Comment:
Could this fail as well for some reason? (I can't think of any)

Reply:
This internally calls `repo.index.commit()`, cf. https://github.com/projectsyn/commodore/blob/660a48a07c810ed9ccc0327c5ed1bbde422619af/commodore/git.py#L190-L199

According to the [Git-Python documentation](https://gitpython.readthedocs.io/en/stable/reference.html#git.index.base.IndexFile.commit) `repo.index.commit()` can't fail.